src/librbd/api/Snapshot.cc

```diff
@@ -309,6 +311,33 @@ int Snapshot<I>::exists(I *ictx, const cls::rbd::SnapshotNamespace& snap_namespa
   return 0;
 }
 
+template <typename I>
+int Snapshot<I>::create(I *ictx, const char *snap_name, uint32_t flags,
+                        ProgressContext& pctx) {
+  ldout(ictx->cct, 20) << "snap_create " << ictx << " " << snap_name
+                       << " flags: " << flags << dendl;
+
+  uint64_t internal_flags = 0;
+
+  if (flags & RBD_SNAP_CREATE_SKIP_QUIESCE) {
+    internal_flags |= SNAP_CREATE_FLAG_SKIP_NOTIFY_QUIESCE;
+    flags &= ~RBD_SNAP_CREATE_SKIP_QUIESCE;
+  }
+  if (flags & RBD_SNAP_CREATE_IGNORE_QUIESCE_ERROR) {
```

Comment:
Nit: `else if`? That will allow it to hit the invalid flag check below since it doesn't make sense to skip quiesce + ignore quiesce errors.